interpreter/cling/lib/Interpreter/ExternalInterpreterSource.cpp

```diff
@@ -0,0 +1,302 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:  Elisavet Sakellari <elisavet.sakellari@cern.ch>
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+#include "ExternalInterpreterSource.h"
+#include "cling/Interpreter/Interpreter.h"
+
+#include "clang/Lex/Preprocessor.h"
+
+using namespace clang;
+
+namespace {
+  class ClingASTImporter : public ASTImporter {
+  private:
+  cling::ExternalInterpreterSource& m_Source;
+
+  public:
+    ClingASTImporter(ASTContext &ToContext, FileManager &ToFileManager,
+                     ASTContext &FromContext, FileManager &FromFileManager,
+                     bool MinimalImport, cling::ExternalInterpreterSource& source):
+      ASTImporter(ToContext, ToFileManager, FromContext, FromFileManager,
+                  MinimalImport), m_Source(source) {}
+
+    Decl *Imported(Decl *From, Decl *To) override {
+      ASTImporter::Imported(From, To);
+
+      if (isa<TagDecl>(From)) {
```

Comment:
Can we do the `isa` on the To? It's the one we use below - which allows us to write this as `if (clang::TagDecl* toTagDecl = dyn_cast<TagDecl>(To)) {`